imap/search_xapian.c

```diff
@@ -2661,9 +2714,16 @@ static uint32_t first_unindexed_uid(search_text_receiver_t *rx)
 
 static int is_indexed_cb(const conv_guidrec_t *rec, void *rock)
 {
-    xapian_update_receiver_t *tr = rock;
+    xapian_update_receiver_t *tr = ((struct is_indexed_rock*)rock)->tr;
+    char doctype = ((struct is_indexed_rock*)rock)->doctype;
 
-    if (rec->part) return 0;
+    if (doctype == XAPIAN_WRAP_DOCTYPE_MSG && rec->part) return 0;
+
+    /* Is this a part in the message we are just indexing? */
+    if (doctype == XAPIAN_WRAP_DOCTYPE_PART && rec->uid == tr->super.uid &&
+            !strcmp(rec->mboxname, tr->super.mailbox->name)) {
```

Comment:
I guess this is OK... I'd prefer to number compare on rec->foldernum, but this is fine.

Question: is this supposed to stop the *P* key being created as well, or make the attachment text not be indexed along with the main message itself?

Reply:
It's supposed to prevent a P-key being created more than once (along with its indexed terms). The attachment text will be indexed along with the G key multiple times (once for each message).